### PR TITLE
os/include/tinyara/lcd: Update init commands for lcd st7785

### PR DESCRIPTION
**Pull request by anjana348**
Gamma settings updated with new init sequence
orientation updated to match LCD refresh direction

### CODE REVIEW

File: os/include/tinyara/lcd/st7785.h [hunk 1]
```diff
@@ -45,13 +45,17 @@
 #if defined(CONFIG_LCD_PORTRAIT) || defined(CONFIG_LCD_RPORTRAIT)
 #define LCD_XRES CONFIG_LCD_XRES
 #define LCD_YRES CONFIG_LCD_YRES
-#define LCD_CASET 0x00EF
-#define LCD_RASET 0x013F
+#define LCD_CASET1 0x00
```
Comment (edwakuwaku) [inline os/include/tinyara/lcd/st7785.h:48]: Could you add more comments to the setting?
Such as below
/* MV = 0, Column = 0x00EF = (240 - 1), Row = 0x013F = (320 - 1) */

File: os/include/tinyara/lcd/st7785.h [hunk 2]
```diff
@@ -45,13 +45,17 @@
...
+#define LCD_CASET2 0xEF
+#define LCD_RASET1 0x01
+#define LCD_RASET2 0x3F
 #else
 #define LCD_XRES CONFIG_LCD_YRES
 #define LCD_YRES CONFIG_LCD_XRES
-#define LCD_CASET 0x013F
-#define LCD_RASET 0x00EF
+#define LCD_CASET1 0x01
```
Comment (edwakuwaku) [inline os/include/tinyara/lcd/st7785.h:55]: /* MV = 1, Column = 0x013F = (320 - 1), Row = 0x00EF = (240 - 1) */

File: os/include/tinyara/lcd/st7785.h (hunk 1 above)
Comment (edwakuwaku) [reply os/include/tinyara/lcd/st7785.h:48]: It will help user to understand those meaning easier

File: os/include/tinyara/lcd/st7785.h [hunk 3]
```diff
@@ -61,30 +65,30 @@
...
-        {0xE1, 28, {0x00, 0xF0, 0x00, 0x0A, 0x00, 0x0E, 0x00, 0x0F, 0x00, 0x0B, 0x00, 0x26, 0x00, 0x2E, 0x00, 0x43, 0x00, 0x45, 0x00, 0x36, 0x00, 0x12, 0x00, 0x12, 0x00, 0x2A, 0x00, 0x32}},
-        {0x2A, 6, {0x00, 0x00, 0x00, 0x00, LCD_CASET && 0xFF00, LCD_CASET && 0x00FF}},//for landscape it is 0x13F and portrait , it is 0xEF
-        {0x2B, 6, {0x00, 0x00, 0x00, 0x00, LCD_RASET && 0xFF00, LCD_RASET && 0x00FF}},//EF changed to 13F as given in datasheet
+        {0xD0, 4, {0x00, 0xA4, 0x00, 0xA1}},
+        {0xE0, 28, {0x00, 0xF0, 0x00, 0x01, 0x00, 0x08, 0x00, 0x05, 0x00, 0x07, 0x00, 0x15, 0x00, 0x32, 0x00, 0x44, 0x00, 0x49, 0x00, 0x37, 0x00, 0x12, 0x00, 0x11, 0x00, 0x2E, 0x00, 0x33}},
+        {0xE1, 28, {0x00, 0xF0, 0x00, 0x0C, 0x00, 0x10, 0x00, 0x0E, 0x00, 0x0C, 0x00, 0x06, 0x00, 0x32, 0x00, 0x33, 0x00, 0x49, 0x00, 0x35, 0x00, 0x10, 0x00, 0x12, 0x00, 0x2C, 0x00, 0x34}},
+        {0x2A, 8, {0x00, 0x00, 0x00, 0x00, 0x00, LCD_CASET1, 0x00, LCD_CASET2}},
+        {0x2B, 8, {0x00, 0x00, 0x00, 0x00, 0x00, LCD_RASET1, 0x00, LCD_RASET2}},
```
Comment (sunghan-chang) [inline os/include/tinyara/lcd/st7785.h:91]: As @edwakuwaku left a comment, how about adding comments to explain what each line means?

**Review by ewoodev — APPROVED** [review]

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**